### PR TITLE
improves bracket undo behavior when vscode autocloses brackets

### PR DESCRIPTION
**Pull request by xconverge**
Yay! We love PRs! 🎊

Please include a description of your change & check your PR against this list, thanks:

- [X] Commit message has a short title & issue references
- [X] Each commit does a logical chunk of work.  
- [X] It builds and tests pass (e.g `gulp tslint`)

More info can be found by clicking the "guidelines for contributing" link above.

### CODE REVIEW

File: src/mode/modeHandler.ts [hunk 1]
```diff
@@ -1095,8 +1096,25 @@ export class ModeHandler implements vscode.Disposable {
...
           break;
-        default:
-          return false;
+      }
+
+      // This section is mostly for vscode auto closing brackets without a user inserting them
+      const letterToTheRight = TextEditor.getLineAt(vimState.cursorPosition).text[vimState.cursorPosition.character];
+      if (letterToTheRight !== undefined) {
+        switch (key) {
+          case "{":
```
Comment (johnfn) [inline src/mode/modeHandler.ts]: Generally when there are giant switch statements that's a sign that you could better represent your stuff as a data structure. In this case, maybe something like this:

```
const pairings = { 
    "{" : "}",
    "[" : "]",
    "<" : ">",

    /* etc... */
}
```

Comment (johnfn) [inline src/mode/modeHandler.ts]: I think this idea would also let you DRY up this code a little bit. It's a little repetitive right now.

**Comment by xconverge** [comment]
Ok definitely a _little_ cleaner now

**Comment by johnfn** [comment]
Do you think you can leverage this?

https://github.com/VSCodeVim/Vim/blob/master/src/matching/matcher.ts

**Comment by xconverge** [comment]
I should be able to... this looks perfect...

**Comment by johnfn** [comment]
Wonderful, thank you!

**Pull request merged**